src/openassetio-python/package/openassetio/test/manager/apiComplianceSuite.py

```diff
@@ -708,6 +711,263 @@ def __testGetWithRelationshipsError(
+
+        actual = self.__concat_all_pages(pager)
+        self.assertListEqual(actual, expected)
+
+    def test_when_result_trait_set_supplied_then_filtered_refs_returned(self):
+        a_ref = self.requireEntityReferenceFixture("a_reference", skipTestIfMissing=False)
+        a_rel = TraitsData(self.requireFixture("a_relationship_trait_set"))
+        result_trait_set = self.requireFixture("an_entity_trait_set_to_filter_by")
+        expected = self.requireEntityReferencesFixture("expected_related_entity_references")
+
+        [pager] = self.__test_getWithRelationshipPaged_success(
+            [a_ref], a_rel, resultTraitSet=result_trait_set
+        )
+
+        actual = self.__concat_all_pages(pager)
+        self.assertListEqual(actual, expected)
+
+    def test_when_querying_missing_reference_then_resolution_error_is_returned(self):
+        self.__test_getWithRelationshipPaged_error(
+            entity_reference=self.requireEntityReferenceFixture(
+                "a_reference_to_a_missing_entity", skipTestIfMissing=False
+            ),
+            relationship_trait_set=self.requireFixture("a_relationship_trait_set"),
+            expected_error_code=BatchElementError.ErrorCode.kEntityResolutionError,
+            expected_error_message=self.requireFixture("expected_error_message"),
+        )
+
+    def test_when_querying_malformed_reference_then_malformed_reference_error_is_returned(self):
+        self.__test_getWithRelationshipPaged_error(
+            entity_reference=self.requireEntityReferenceFixture(
+                "a_malformed_reference", skipTestIfMissing=False
+            ),
+            relationship_trait_set=self.requireFixture("a_relationship_trait_set"),
+            expected_error_code=BatchElementError.ErrorCode.kMalformedEntityReference,
+            expected_error_message=self.requireFixture("expected_error_message"),
+        )
+
+    def test_when_related_entities_span_multiple_pages_then_pager_has_multiple_pages(self):
+        a_ref = self.requireEntityReferenceFixture("a_reference", skipTestIfMissing=False)
+        a_rel = TraitsData(self.requireFixture("a_relationship_trait_set"))
+        expected_related_refs = self.requireEntityReferencesFixture(
+            "expected_related_entity_references"
+        )
+
+        # Ensure we have at least two related references, so that we can
+        # ensure at least two pages given a page size of 1.
+        self.assertGreater(
+            len(expected_related_refs),
+            1,
+            msg="Please provide fixtures that result in at least two related references",
+        )
+
+        for page_size in (1, 2):
+            with self.subTest(page_size=page_size):
+                # Split expected related references list into pages.
+                expected_pages = [
+                    expected_related_refs[page_start : page_start + page_size]
+                    for page_start in range(0, len(expected_related_refs), page_size)
```

Comment:
[consider] Sorry if this is doing this, is it worth deliberately requesting one page past the end to help assert the "does not error if you go past the end" part of the pager contract? [edit] I just saw you might already be doing this in the next commit refactor

Reply:
Yeah, the `self._assert_pager_is_at_end(pager)` checks this case.